language/move-prover/spec-lang/src/env.rs

```diff
@@ -1443,10 +1464,27 @@ impl<'env> ModuleEnv<'env> {
     pub fn get_node_instantiation(&self, node_id: NodeId) -> Vec<Type> {
         self.data
             .instantiation_map
+            .borrow()
             .get(&node_id)
             .cloned()
             .unwrap_or_else(Vec::new)
     }
+
+    /// Add type parameter associated with the given node.
+    pub fn set_node_instantiation(&self, node_id: &NodeId, typ: Type) {
+        let new_types = match self.data.instantiation_map.borrow().get(node_id) {
+            Some(types) => {
```

Comment:
This function has a misleading name. It does not set the instantiation, but adds a type to it. Perhaps `extend_node_instantiation`?